font/freetype: disable SVG glyphs, simplify color check

qwerasd205 opened the pull request:
We don't currently support rendering SVG glyphs so they should be ignored when loading. Additionally, the check for whether a glyph is colored has been simplified by just checking the pixel mode of the rendered bitmap.

This commit also fixes a bug caused by calling the color check inside of `renderGlyph`, which caused the bitmap to be freed creating a chance for memory corruption and garbled glyphs. This bug was introduced by 40c1140 (#6602) and discussed in #6781.

qwerasd205 committed:
font/freetype: disable SVG glyphs, simplify color check

We don't currently support rendering SVG glyphs so they should be
ignored when loading. Additionally, the check for whether a glyph is
colored has been simplified by just checking the pixel mode of the
rendered bitmap.

This commit also fixes a bug caused by calling the color check inside of
`renderGlyph`, which caused the bitmap to be freed creating a chance for
memory corruption and garbled glyphs.

mitchellh reviewed (approved):
The color detection stuff is a bit spooky. It makes sense but I wonder why my past self didn't do this haha. Approving we can find regressions in test.

qwerasd205 committed:
fix(font/shape): don't require emoji presentation for grapheme parts

Also update shaper test that fails because the run iterator can't apply
that logic since `testWriteString` doesn't do proper grpaheme clustering
so the parts are actually split across multiple cells.

Several other tests are technically incorrect for the same reason but
still pass, so I've decided not to fix them here.

qwerasd205 commented:
Fixed the test failure -- this whole PR is fairly spooky so we definitely do need to watch out for regressions.

mitchellh commented on src/font/face/freetype.zig:
```diff
@@ -270,25 +270,21 @@ pub const Face = struct {
 
-        // If the glyph is SVG we assume colorized
         const glyph = self.face.handle.*.glyph;
-        if (glyph.*.format == freetype.c.FT_GLYPH_FORMAT_SVG) return true;
```
I realized that on distributed Ghostty, we dynamically link to freetype so SVG support may actually work. I'm not sure what this means for us but want to mention it here since you're more aware.

mitchellh reviewed (approved):
Looks good and I approve. I made one note about SVG support.

qwerasd205 commented:
As far as I know there is no way to statically configure FreeType to support SVG, see https://freetype.org/freetype2/docs/reference/ft2-svg_fonts.html - the programmer has to provide hooks when initializing FreeType within their program.

mitchellh commented:
Ah I see. Well, very good then let's give it a go! Thanks
